parachain/pallets/outbound-queue/src/send_message_impl.rs

```diff
@@ -57,7 +58,14 @@ where
 			.unwrap_or_else(|| unique((message.channel_id, &message.command)).into());
 
 		let gas_used_at_most = T::GasMeter::maximum_gas_used_at_most(&message.command);
+		ensure!(
+			gas_used_at_most > T::GasMeter::MAXIMUM_BASE_GAS &&
+				gas_used_at_most < T::GasMeter::MAXIMUM_GAS_CAP,
+			SendError::MessageInvalidGasFees
+		);
 		let fee = Self::calculate_fee(gas_used_at_most, T::PricingParameters::get());
+		let zero_fee = <Self as SendMessageFeeProvider>::Balance::zero();
+		ensure!(fee.local > zero_fee && fee.remote > zero_fee, SendError::MessageInvalidGasFees);
```

Comment:
This check is for the test case as I mentioned here in https://github.com/Snowfork/snowbridge/pull/1109#discussion_r1453721841 in which case the `remote_fee` calculated is zero which leads to the sovereign account of the sibling chain is not properly refunded. Finally we can't fund the agent on the Ethereum side.

`MAXIMUM_GAS_CAP` added here is a hard limit that applies to all commands. It's mainly for command [Upgrade](https://github.com/Snowfork/snowbridge/blob/d77201bc370ab2a4da15de701871b4a643bd5b5c/parachain/primitives/core/src/outbound.rs#L397) with a sanity dynamic gas. Otherwise the gas check in [Gateway](https://github.com/Snowfork/snowbridge/blob/d77201bc370ab2a4da15de701871b4a643bd5b5c/contracts/src/Gateway.sol#L152-L154) will fail and the bridge stuck. It also applies to the future command arbitrary transact the gas is dynamic from input of the end-user.

Reply:
`MAXIMUM_GAS_CAP` added here is a hard limit that applies to all commands. It's mainly for command [Upgrade](https://github.com/Snowfork/snowbridge/blob/d77201bc370ab2a4da15de701871b4a643bd5b5c/parachain/primitives/core/src/outbound.rs#L397) with a sanity dynamic gas. It also applies to the future command arbitrary transact for which the gas is dynamic from the input of the end-user. Otherwise, the gas check in [Gateway](https://github.com/Snowfork/snowbridge/blob/d77201bc370ab2a4da15de701871b4a643bd5b5c/contracts/src/Gateway.sol#L152-L154) will always fail and the bridge stuck.

Reply:
I don't think we should introduce `MAXIMUM_GAS_CAP`, for the same reasons I outlined in https://github.com/Snowfork/snowbridge/pull/1109#discussion_r1453336525.

Reply:
See https://github.com/Snowfork/snowbridge/pull/1109#discussion_r1455250041. I think its fine if calculated remote fee is zero. It just means the fee is too low to be represented using DOT decimals.

Reply:
Checks reverted in https://github.com/Snowfork/snowbridge/pull/1109/commits/af15053f0d01cfa7f06bd6f3119987873793297a